algebird-core/src/main/scala/com/twitter/algebird/RingBuf.scala

```diff
@@ -0,0 +1,105 @@
+package com.twitter.algebird
+
+/**
+ * Value class to hold references to indices internal to RingBuf.
+ */
+case class Idx(value: Int) extends AnyVal {
```

Comment:
what do y'all think of this? Might be goofy. It's not doing much other than enforcing its own type.